modules/javafx.controls/src/main/java/javafx/scene/control/skin/ListViewSkin.java

```diff
@@ -356,9 +356,7 @@ public ListViewSkin(final ListView<T> control) {
         flow.setCellCount(newCount);
 
         updatePlaceholderRegionVisibility();
-        if (newCount != oldCount) {
-            requestRebuildCells();
-        } else {
+        if (newCount == oldCount) {
```

Comment:
Does this also need the same `else if (oldCount == 0)` test that you added to `TableViewSkinBase`?

Reply:
`TableViewSkinBase` has an `itemsChangeListener` that triggers the `rowCountListener`, which sets `itemCount = 0` when an item has been replaced. `ListViewSkin` uses a different approach via `listViewItemsListener` -> `flow::setCellDirty`.

However, the same approach (`itemCount = 0`) is used in both cases when the list of items is cleared, so I'll add it here as well.